fire/core.py

```diff
@@ -730,15 +732,17 @@ def _ParseKeywordArgs(args, fn_args, fn_keywords):
         if got_argument and (keyword in fn_args or fn_keywords):
           kwargs[keyword] = value
           skip_argument = not contains_equals and not is_bool_syntax
-          arg_consumed = True
+        else:
+          unconsumed_named_args.append(argument)
+        arg_consumed = True
```

Comment:
We can remove `arg_consumed` altogether (here and at 698), and replace the if at 739 with an `else`.